src/client/Microsoft.Identity.Client/OAuth2/Throttling/IThrottlingProvider.cs

```diff
@@ -0,0 +1,25 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System.Collections.Generic;
+using Microsoft.Identity.Client.Internal.Requests;
+
+namespace Microsoft.Identity.Client.OAuth2.Throttling
+{
+    internal interface IThrottlingProvider
+    {
+        void TryThrottle(
+            AuthenticationRequestParameters requestParams, 
+            IReadOnlyDictionary<string, string> bodyParams);
```

Comment:
nit: you don't use any other values in requestParams, just the Logger, could you pass just the logger directly? or maybe it doesn't make a big enough difference.